src/taco-cli/test/kit.ts

```diff
@@ -233,16 +348,29 @@ describe("Kit", function (): void {
             rimraf(cliProjectpath, function (err: Error): void { done(); }); // ignore errors
         });
 
-        it("'taco kit select --kit {kit-ID}' should execute with no errors", function (done: MochaDone): void {
-            runKitCommandAndVerifyTacoJsonContents(["select", "--kit", "5.1.1-Kit"], tacoJsonPath, expectedKitTacoJsonKeyValues)
-                .then((telemetryParameters: TacoUtility.ICommandTelemetryProperties) => {
-                    var expected: TacoUtility.ICommandTelemetryProperties = {
+        it("'taco kit select --kit {Invalid-kit-ID}' should execute with expected error", function (done: MochaDone): void {
+            runKitCommandFailureCaseAndVerifyTacoJson<TacoKitsErrorCodes>(["select", "--kit", "InvalidKit"], tacoJsonPath, expectedCliTacoJsonKeyValues2, TacoKitsErrorCodes.TacoKitsExceptionInvalidKit)
```

Comment:
A better approach to tests which succeed or fail as a promise chain is to just return the promise, and not accept a done function. Just found this out recently, but it cleans things up a bunch.

This would then be
`it("...", function (): Q.promise<any> { return runKitCommandCaseAndVerifyTacoJson(...);})`